src/sound_openal.cpp

```diff
@@ -175,14 +164,104 @@ SoundBuffer* loadOggFile(const std::string &filepath)
 				<<"preparing sound buffer"<<std::endl;
 	}
 
-	infostream<<"Audio file "<<filepath<<" loaded"<<std::endl;
+	infostream << "Audio file "
+		<< filename_for_logging << " loaded" << std::endl;
 
 	// Clean up!
-	ov_clear(&oggFile);
+	ov_clear(oggFile);
 
 	return snd;
 }
 
+SoundBuffer *load_ogg_from_file(const std::string &path)
+{
+	OggVorbis_File oggFile;
+
+	// Try opening the given file.
+	// This requires libvorbis >= 1.3.2, as
+	// previous versions expect a non-const char *
+	if (ov_fopen(path.c_str(), &oggFile) != 0) {
+		infostream << "Audio: Error opening " << path
+			<< " for decoding" << std::endl;
+		return NULL;
+	}
+
+	return load_opened_ogg_file(&oggFile, path);
+}
+
+struct buffer_sound_t {
+	const char *buf;
+	size_t cur_offset;
+	size_t len;
+};
+
+size_t buffer_sound_read_func(void *ptr, size_t size, size_t nmemb, void *datasource)
+{
+	buffer_sound_t *s = (buffer_sound_t *)datasource;
+	size_t copied_size = MYMIN(s->len - s->cur_offset, size);
+	memcpy(ptr, s->buf + s->cur_offset, copied_size);
+	s->cur_offset += copied_size;
+	return copied_size;
+}
+
+int buffer_sound_seek_func(void *datasource, ogg_int64_t offset, int whence)
+{
+	buffer_sound_t *s = (buffer_sound_t *)datasource;
+	if (whence == SEEK_SET) {
+		if (offset < 0 || (size_t)MYMIN(offset, 0) >= s->len) {
```

Comment:
Er, doesn't the `MYMIN` clamp the value to the range `(-∞, 0]`?  And since you've already checked negative values that would just be `{0}`.